### PR TITLE
Added support for facts module for Dell Networking OS10 device.

### PR DESCRIPTION
**Pull request by skg-net**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME
- modules/core/network/dnos10/dnos10_facts
##### ANSIBLE VERSION
- ansible 2.2.0 (devel f497d771c8) last updated 2016/09/15 12:00:53 (GMT -700)
  lib/ansible/modules/core: (detached HEAD 48d932643b) last updated 2016/09/15 12:01:19 (GMT -700)
  lib/ansible/modules/extras: (detached HEAD aa45bd8a94) last updated 2016/09/15 12:01:19 (GMT -700)
  config file = /home/skg/lothlorien/ansible.cfg
  configured module search path = Default w/o overrides
##### SUMMARY
- Added support for facts module for Dell Networking OS10 device.

### CODE REVIEW

**Comment by skg-net** [comment]
@gundalow @privateip , Please have a look at it.

**Comment by gregdek** [comment]
Thanks again to @skg-net. This module is going into community review.

We notice that you are one of the maintainers (@dhivyap, @skg-net, @privateip, @gundalow) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

File: network/dnos10/dnos10_facts.py [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+"""
+
+import re
+import itertools
+
+from ansible.module_utils.basic import get_exception
+from ansible.module_utils.netcli import CommandRunner
+from ansible.module_utils.network import NetworkModule
+import ansible.module_utils.dnos10
+import xml.etree.ElementTree as ET
```
Comment (privateip) [inline network/dnos10/dnos10_facts.py]: need to wrap this in a try ... except block as its not part of the std lib.  see [here](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/junos.py#L52) for example

**Review by privateip — CHANGES_REQUESTED** [review]

**Comment by gundalow** [comment]
needs_revision - As per Peters comments above

**Comment by gregdek** [comment]
Thanks @skg-net for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

File: network/dnos10/dnos10_facts.py (hunk 1 above)
Comment (skg-net) [reply network/dnos10/dnos10_facts.py]: xml.etree.ElementTree is part of standard library.  I update the code as per your request.

**Comment by skg-net** [comment]
ready_for_review

File: network/dnos10/dnos10_facts.py (hunk 1 above)
Comment (gundalow) [reply network/dnos10/dnos10_facts.py]: `from lxml import etree as ET` reports `No module named lxml` on Python 2.6.6.
`xml.etree.ElementTree` does work on Python 2.6.6

Comment (skg-net) [reply network/dnos10/dnos10_facts.py]: need to install lxml, its not part of the standard package

**Comment by gundalow** [comment]
Merged, thanks.

I believe this is the last PR so you can go ahead and commence the renaming.

**Pull request merged**